### ISSUE TITLE
OpenAPI changes broke something

### ISSUE THREAD

**Issue by cscetbon** [Orange-OpenSource/casskop#166]
We're no longer able to deploy a cluster if we use the new crd with the validation included. As a temporary workaround until it's fixed (hopefully soon by @jsanda 😉) what can be done is to checkout an older version of the crd and install it instead of the latest version :
```
git checkout bca1db27 -- deploy/crds/db_v1alpha1_cassandracluster_crd.yaml
kubectl apply -f deploy/crds/db_v1alpha1_cassandracluster_crd.yaml
kubectl apply -f test/e2e/testdata/cassandracluster-1DC.yaml
```

**Comment by jsanda** [comment]
Hey @cscetbon 

I was not able to reproduce the issue. I built from master, my HEAD is at `4e961fad8`. I was able to deploy to a kind cluster. Here are my steps:

* `make build`
* `kubectl apply -f deploy/crds/db_v1alpha1_cassandracluster_crd.yaml`
* `make run`
* `kubectl apply -f cluster.yaml`

where cluster.yaml looks like:

```yaml
apiVersion: "db.orange.com/v1alpha1"
kind: "CassandraCluster"
metadata:
  name: cluster1
spec:
  dataCapacity: 10Gi
  imagepullpolicy: IfNotPresent
  deletePVC: false
  maxPodUnavailable: 0
  resources:
    requests:
      cpu: '2'
      memory: 1Gi
    limits:
      cpu: '2'
      memory: 1Gi
  topology:
    dc:
      - name: dc1
        rack:
          - name: rack1
```
Can you test again and please provide steps to reproduce if there is still an issue?

Thanks

John

**Comment by cscetbon** [comment]
@jsanda agreed. I think my k8s was in bad state cause I can't reproduce it either. I created a branch to fix the issue with the required fields #165
